install/operator/pkg/syndesis/action/install.go

```diff
@@ -141,22 +138,25 @@ func (a *installAction) Execute(ctx context.Context, syndesis *v1beta1.Syndesis)
 
 	// Render the database resource if needed...
 	if syndesis.Spec.Components.Database.ExternalDbURL == "" {
-		dbResources, err := generator.RenderDir("./database/", config)
-		if err != nil {
-			return err
-		}
+		// TODO: Remove for 7.7. Migrate database upgrade to a step in upgrade process
+		if syndesisPhaseIs(syndesis, v1beta1.SyndesisPhaseInstalling) || syndesisPhaseIs(syndesis, v1beta1.SyndesisPhasePostUpgradeRun) {
```

Comment:
I think due to polling iterations happening so quickly/frequently, install.go is run through 100s of times every minute ->

> 15:05:09 -> New operator pushed to namespace
> ...            -> 5863 iterations of loop in phase Starting !! 
> 15:10:52 -> Executing Upgrading phase

As the db assets are rendered and installed as a new deployment config, presumably a race condition must occur. The deployment config is checked a number of times before an install actually occurs. The deployment config is updated twice before a deployment and the UPGRADE env var is never deployed.

Reply:
2nd Possibility ...

Changing this file is no longer necessary. Since the problem has been updating the DeploymentConfig twice then stop the 2nd change-back so modify instead `configuration.GetProperties` :
```
databaseDeployment := &appsv1.DeploymentConfig{}
		if err := client.Get(ctx, types.NamespacedName{Namespace: syndesis.Namespace, Name: "syndesis-db"}, databaseDeployment); err == nil {
			for _, c := range databaseDeployment.Spec.Template.Spec.Containers {
				if c.Name == "postgresql" {
					//
					// Does deploment config already contain UPGRADE Env Var?
					// if it does then DO NOT remove it.
					//
					hasUpgradeEnvVar := false
					for _, env := range c.Env {
						if env.Name == "POSTGRESQL_UPGRADE" {
							hasUpgradeEnvVar = true
						}
					}

					configuration.DatabaseNeedsUpgrade = hasUpgradeEnvVar || c.Image != configuration.Syndesis.Components.Database.Image
					break
				}
			}
		}
```